src/main/java/com/google/ehub/utility/NextIncreasingLexicographicStringUtility.java

```diff
@@ -0,0 +1,37 @@
+package com.google.ehub.utility;
+
+/**
+ * Utility class that holds method to find the next lexicographic string.
+ */
+public final class NextIncreasingLexicographicStringUtility {
+  /**
+   * Finds the next increasing lexicographic string.
+   *
+   * @param str the string used to find the next increasing lexicographic string
+   * @return the next lexicographic string in increasing order
+   */
+  public static String getNextIncreasingLexicographicString(String str) {
+    if (str == null) {
+      return "";
+    }
+    if (str.isEmpty()) {
+      return "a";
+    }
+
+    int i = str.length() - 1;
+
+    while (i >= 0 && str.charAt(i) == 'z') {
+      --i;
+    }
+
+    if (i == -1) {
+      str += 'a';
+    } else {
+      str = str.substring(0, i) + (char) ((int) str.charAt(i) + 1) + str.substring(i + 1);
```

Comment:
I think readability would actually be improved if you broke this down:

```
char nextCharAtIndex = str.charAt(i) + 1
// some comment about returning the string with just a modified char
return str.substring(0, i) + nextCharAtIndex + str.substring(i + 1);
```
in general also I think it'd be good to have more documentation on what you're doing in this method

Reply:
I added more detailed documentation in last commit.